packages/language-server/src/ui5-model.ts

```diff
@@ -170,58 +267,24 @@ function getTypeNameFix(): TypeNameFix {
   return fixes;
 }
 
-function getLibs(): string[] {
-  // When we support more version the following libraries should be added:
-  // "sap.fileviewer"
-  // "sap.ui.testrecorder"
-  // "sap.zen.dsh"
-  // "sap.zen.crosstab"
-  return [
-    "sap.ui.core",
-    "sap.apf",
-    "sap.ca.scfld.md",
-    "sap.ca.ui",
-    "sap.chart",
-    "sap.collaboration",
-    "sap.f",
-    "sap.fe",
-    "sap.gantt",
-    "sap.landvisz",
-    "sap.m",
-    "sap.makit",
-    "sap.me",
-    "sap.ndc",
-    "sap.ovp",
-    "sap.rules.ui",
-    "sap.suite.ui.commons",
-    "sap.suite.ui.generic.template",
-    "sap.suite.ui.microchart",
-    "sap.tnt",
-    "sap.ui.codeeditor",
-    "sap.ui.commons",
-    "sap.ui.comp",
-    "sap.ui.dt",
-    "sap.ui.export",
-    "sap.ui.fl",
-    "sap.ui.generic.app",
-    "sap.ui.generic.template",
-    "sap.ui.integration",
-    "sap.ui.layout",
-    "sap.ui.mdc",
-    "sap.ui.richtexteditor",
-    "sap.ui.rta",
-    "sap.ui.suite",
-    "sap.ui.support",
-    "sap.ui.table",
-    "sap.ui.unified",
-    "sap.ui.ux3",
-    "sap.ui.vbm",
-    "sap.ui.vk",
-    "sap.ui.vtm",
-    "sap.uiext.inbox",
-    "sap.ushell",
-    "sap.uxap",
-    "sap.viz",
-    "sap.zen.commons",
-  ];
+async function getLibsAsync(
+  cacheFilePath: string | undefined,
+  cdnBaseUrl: string
+): Promise<string[]> {
+  let libs = (await readFromCache(cacheFilePath)) as string[];
+  if (libs === undefined) {
+    const url = `${cdnBaseUrl}resources/sap-ui-version.json`;
+    const response = await fetch(url);
+    if (response.ok) {
+      const versionInfo = await response.json();
+      // read libraries from version information
+      libs = versionInfo?.libraries?.map((lib) => {
```

Comment:
The API.json is in a state where all the libs there are fully valid?
No filtering / exclusion needed?

Reply:
All invalid libs will not return an api.json which is fine IMO - we can implement another feature which adds an indicator in the sap-ui-version.json to know whether api.json is available or not. I would keep this for a next iteration...